doc/stdenv/meta.chapter.md

```diff
@@ -182,7 +182,9 @@ meta.hydraPlatforms = [];
 
 ### `broken` {#var-meta-broken}
 
-If set to `true`, the package is marked as "broken", meaning that it won’t show up in `nix-env -qa`, and cannot be built or installed. Such packages should be removed from Nixpkgs eventually unless they are fixed.
+If set to `true`, the package is marked as "broken", meaning that it won’t show up in `nix-env -qa`, and cannot be built or installed unless the environment variable `NIXPKGS_ALLOW_BROKEN` is set. Such unconditionally-broken packages should be removed from Nixpkgs eventually unless they are fixed.
+
+The value of this attribute can depend on a package's arguments, including `stdenv`.  This means that `broken` can be used to express constraints such as "does not cross compile" (`meta.broken = !(stdenv.buildPlatform.canExecute stdenv.hostPlatform)`) or "broken if all of a certain set of its dependencies are broken".  This makes `broken` strictly more powerful than `meta.badPlatforms`.  However `meta.availableOn` currently examines only `meta.platforms` and `meta.badPlatforms`, so `meta.broken` does not influence the default values for optional dependencies.
```

Comment:
Please add links to definitions of those code tokens, otherwise one has to search for them manually, and that may simply fail.

Suggestion to make examples stand out better:

````suggestion
The value of this attribute can depend on a package's arguments, including `stdenv`. 
This means that `broken` can be used to express constraints, for example:

- Does not cross compile

  ```nix
   meta.broken = !(stdenv.buildPlatform.canExecute stdenv.hostPlatform)
  ```

- Broken if all of a certain set of its dependencies are broken

  ```nix
  meta.broken = lib.all (map (p: p.meta.broken) [ glibc musl ])
  ```

This makes `broken` strictly more powerful than `meta.badPlatforms`.
However `meta.availableOn` currently examines only `meta.platforms` and `meta.badPlatforms`, so `meta.broken` does not influence the default values for optional dependencies.
````